backend/src/apiserver/server/experiment_server.go

```diff
@@ -328,26 +328,23 @@ func (s *ExperimentServer) canAccessExperiment(ctx context.Context, experimentID
 		// Skip authorization if not multi-user mode.
 		return nil
 	}
-
+	namespace := ""
```

Comment:
I see that the value of `resourceAttributes.Namespace` takes the order of 
1. `resourceAttributes.Namespace` itself if non-empty
2. `experiment.Namespace` if this is non-empty
3. error if both `resourceAttributes.Namespace` and `experiment.Namespace` are empty

I wonder if we should add a verification that `resourceAttributes.Namespace == experiment.Namespace` when they are both provided.

Reply:
Thank you. I agree. This shouldn't happen. Whenever an `experimentId` is present, the `resourceAttributes.Namespace` should be come from the experiment. Let me implement it.